demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1312 @@
+
+/**
+ * @brief Length of path to server certificate.
+ */
+#define ROOT_CA_CERT_PATH_LENGTH    ( ( uint16_t ) ( sizeof( ROOT_CA_CERT_PATH ) - 1 ) )
+
+/**
+ * Provide default values for undefined configuration settings.
+ */
+#ifndef BROKER_PORT
+    #define BROKER_PORT    ( 8883 )
+#endif
+
+#ifndef NETWORK_BUFFER_SIZE
+    #define NETWORK_BUFFER_SIZE    ( 1024U )
+#endif
+
+/**
+ * @brief Length of client identifier.
+ */
+#define CLIENT_IDENTIFIER_LENGTH                ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) - 1 ) )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milli seconds.
+ */
+#define CONNACK_RECV_TIMEOUT_MS                 ( 1000U )
+
+/**
+ * @brief The MQTT topic filter to subscribe to for temperature data in the demo.
+ */
+#define DEMO_TEMPERATURE_TOPIC_FILTER           CLIENT_IDENTIFIER "/demo/temperature/+"
+
+/**
+ * @brief The length of the temperature topic filter.
+ */
+#define DEMO_TEMPERATURE_TOPIC_FILTER_LENGTH    ( ( uint16_t ) ( sizeof( DEMO_TEMPERATURE_TOPIC_FILTER ) - 1U ) )
+
+/**
+ * @brief The MQTT topic for the high temperature data value that the demo will
+ * publish to.
+ */
+#define DEMO_TEMPERATURE_HIGH_TOPIC             CLIENT_IDENTIFIER "/demo/temperature/high"
+
+/**
+ * @brief The length of the high temperature topic name.
+ */
+#define DEMO_TEMPERATURE_HIGH_TOPIC_LENGTH      ( ( uint16_t ) ( sizeof( DEMO_TEMPERATURE_HIGH_TOPIC ) - 1U ) )
+
+/**
+ * @brief The MQTT topic for the high temperature data value that the demo will
+ * publish to.
+ */
+#define DEMO_TEMPERATURE_LOW_TOPIC              CLIENT_IDENTIFIER "/demo/temperature/low"
+
+/**
+ * @brief The length of the high temperature topic name.
+ */
+#define DEMO_TEMPERATURE_LOW_TOPIC_LENGTH       ( ( uint16_t ) ( sizeof( DEMO_TEMPERATURE_LOW_TOPIC ) - 1U ) )
+
+/**
+ * @brief The MQTT topic filter to subscribe and publish to for humidity data in the demo.
+ *
+ * The topic name starts with the client identifier to ensure that each demo
+ * interacts with a unique topic name.
+ */
+#define DEMO_HUMIDITY_TOPIC                     CLIENT_IDENTIFIER "/demo/humidity"
+
+/**
+ * @brief The length of the humidity topic.
+ */
+#define DEMO_HUMIDITY_TOPIC_LENGTH              ( ( uint16_t ) ( sizeof( DEMO_HUMIDITY_TOPIC ) - 1U ) )
+
+/**
+ * @brief The MQTT topic filter to subscribe and publish to for precipitation data in the demo.
+ *
+ * The topic name starts with the client identifier to ensure that each demo
+ * interacts with a unique topic name.
+ */
+#define DEMO_PRECIPITATION_TOPIC                CLIENT_IDENTIFIER "/demo/precipitation"
+
+/**
+ * @brief The length of the precipitation topic.
+ */
+#define DEMO_PRECIPITATION_TOPIC_LENGTH         ( ( uint16_t ) ( sizeof( DEMO_PRECIPITATION_TOPIC ) - 1U ) )
+
+/**
+ * @brief The MQTT message for the #DEMO_TEMPERATURE_HIGH_TOPIC topic published
+ * in this example.
+ */
+#define DEMO_TEMPERATURE_HIGH_MESSAGE           "Today's High is 80 degree F"
```

Comment:
Perhaps we should use this opportunity to push Kelvin as the superior unit of measurement

Reply:
Interesting point (to make the demo universally appealing) 😆